statman/Gaussiandistribution.py

```diff
@@ -1,206 +1,203 @@
-		axes[0].set_ylabel('Density')
-		plt.show()
-
-		return x, y
-		
-	def __add__(self, other):
-		
-		"""Function to add together two Gaussian distributions
-		
-		Args:
-			other (Gaussian): Gaussian instance
-			
-		Returns:
-			Gaussian: Gaussian distribution
-			
-		"""
-		
-		result = Gaussian()
-		result.mean = self.mean + other.mean
-		result.stdev = math.sqrt(self.stdev ** 2 + other.stdev ** 2)
-		
-		return result
-		
-		
-	def __repr__(self):
-	
-		"""Function to output the characteristics of the Gaussian instance
-		
-		Args:
-			None
-		
-		Returns:
-			string: characteristics of the Gaussian
-		
-		"""
-		
-		return "mean {}, standard deviation {}".format(self.mean, self.stdev)
\ No newline at end of file
+    """ Gaussian distribution class for calculating and 
+    visualizing a Gaussian distribution.
+
+    Attributes:
+            mean (float) representing the mean value of the distribution
+            stdev (float) representing the standard deviation of the distribution
+            data_list (list of floats) a list of floats extracted from the data file
+
+    """
+
+    def __init__(self, mu=0, sigma=1):
+
+        statman.__init__(self, mu, sigma)
+
+    def calculate_mean(self):
+        """Function to calculate the mean of the data set.
+
+        Args: 
+                None
+
+        Returns: 
+                float: mean of the data set
+
+        """
+
+        avg = 1.0 * sum(self.data) / len(self.data)
+
+        self.mean = avg
+
+        return self.mean
+
+    def calculate_stdev(self, sample=True):
+        """Function to calculate the standard deviation of the data set.
+
+        Args: 
+                sample (bool): whether the data represents a sample or population
+
+        Returns: 
+                float: standard deviation of the data set
+
+        """
+
+        if sample:
+            n = len(self.data) - 1
+        else:
+            n = len(self.data)
+
+        mean = self.mean
+
+        sigma = 0
+
+        for d in self.data:
+            sigma += (d - mean) ** 2
+
+        sigma = math.sqrt(sigma / n)
+
+        self.stdev = sigma
+
+        return self.stdev
+
+    def read_data(self, file_name, sample=True):
+        """Function to read in data from a txt file. The txt file should have
+        one number (float) per line. The numbers are stored in the data attribute. 
+        After reading in the file, the mean and standard deviation are calculated
+
+        Args:
+                file_name (string): name of a file to read from
+
+        Returns:
+                None
+
+        """
+
+        if isinstance(file_name, str):
+            with open(file_name) as file:
+                data_list = []
+                line = file.readline()
+                while line:
+                    data_list.append(int(line))
+                    line = file.readline()
+            file.close()
+        elif isinstance(file_name, pandas.core.series.Series):
```

Comment:
Might be better to read in a csv or xslx file and then convert to a pandas series before working on it, probably do this:
`if isinstance(file_name, csv):
        df = read_csv(....)
        .....
`

Reply:
Alright, I'll work on that. Cause while doing this, I only had working with one column at a time in mind, but let's see how this goes.